### DISCUSSION TITLE
Serialization of java.util.Map

### DISCUSSION THREAD

**Discussion by chemeugene** [kobylynskyi/graphql-java-codegen#663] [Q&A] [answered]
Hi, 
My input type contains JSON scalar
```
input FeeInput {
    feeType: String!,
    feeParams: JSON!
}
```

graphqlCodegen task has section
```
    customTypesMapping = [
            JSON      : "java.util.Map<java.lang.String,java.lang.String>"
    ]
```

The problem consists of when client sends request it serializes this object like this:
```
{ feeType: \"type\", feeParams: {k1=v1} }
```
As you can see, key/value is unquoted, but my grapqh-java server expects key is unquoted, but value is quoted.

Is there a way to overcome this?

**Comment by robbertnoordzij** [comment]
I think it has something the do with the serialisation of the Map. 
You can configure the object mapper using: GraphQLRequestSerializer.OBJECT_MAPPER

**Comment by kobylynskyi** [comment, marked as answer]
@chemeugene I've created a new issue to track this: #668 and a PR with changes to `GraphQLRequestSerializer`.
Please review and let me know what you think: https://github.com/kobylynskyi/graphql-java-codegen/pull/669/files#diff-09c6dd12f22abc1a50704cc01179bf263e44284d0107352701fb51231828f156R439-R462

**Reply by chemeugene** [reply]
@kobylynskyi  Excelent! I think this is correct solution because approach with configuring GraphQLRequestSerializer.OBJECT_MAPPER only does't work
